File: AuctionServer/AuctionServer.Domain/Entities/AuctionItem.cs
namespace AuctionServer.Domain.Entities;

public class AuctionItem
{
    public Guid Id { get; set; }
    public string Brand { get; set; }
    public string Model { get; set; }
    public int Year { get; set; }
    public int Mileage { get; set; }
    public string Color { get; set; }
    public string Engine { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    
    public string ImageUrl { get; set; }
    
    public Guid AuctionListingId { get; set; }
    public AuctionListing AuctionListing { get; set; }
    
    public ICollection<Bid>? Bids { get; set; }
    
    public Guid? PaymentId { get; set; }
    public Payment? Payment { get; set; }
    
}

File: AuctionClient/src/types/AuctionItem.ts
export interface Item {
  id?: string;
  brand: string;
  model: string;
  year: number;
  mileage: number;
  color: string;
  engine: string;
  description: string;
  attachment: string;
  price: number;
  imageUrl: string;
  auctionListingId: string;
}

export interface UploadItem {
  brand: string;
  model: string;
  year: number;
  mileage: number;
  color: string;
  engine: string;
  description: string;
  attachment: string;
  price: number;
  image: File | null;
  auctionListingId: string;
}
